src/compiler/crystal/codegen/link.cr

```diff
@@ -105,83 +110,68 @@ module Crystal
     end
 
     private def lib_flags_posix
-      library_path = ENV["LIBRARY_PATH"]?.try(&.split(':', remove_empty: true)) ||
-                     ["/usr/lib", "/usr/local/lib"]
-      has_pkg_config = nil
+      flags = [] of String
+      static_build = has_flag?("static")
 
-      String.build do |flags|
-        link_annotations.reverse_each do |ann|
-          if ldflags = ann.ldflags
-            flags << ' ' << ldflags
-          end
+      # Instruct the linker to link statically if the user asks
+      flags << "-static" if static_build
 
-          if libname = ann.lib
-            if has_pkg_config.nil?
-              has_pkg_config = Process.run("which", {"pkg-config"}, output: Process::Redirect::Close).success?
-            end
-
-            static = has_flag?("static") || ann.static?
-
-            if static && (static_lib = find_static_lib(libname, CrystalLibraryPath.paths))
-              flags << ' ' << static_lib
-            elsif has_pkg_config && (libflags = pkg_config_flags(libname, static, library_path))
-              flags << ' ' << libflags
-            elsif static && (static_lib = find_static_lib(libname, library_path))
-              flags << ' ' << static_lib
-            else
-              flags << " -l" << libname
-            end
-          end
+      # Add CRYSTAL_LIBRARY_PATH locations, so the linker preferentially
+      # searches user-given library paths.
+      CrystalLibraryPath.paths.each do |path|
+        flags << "'-L#{path}'"
+      end
 
-          if framework = ann.framework
-            flags << " -framework " << framework
-          end
+      link_annotations.reverse_each do |ann|
+        if ldflags = ann.ldflags
+          flags << ldflags
         end
 
-        # Append the CRYSTAL_LIBRARY_PATH values as -L flags.
-        CrystalLibraryPath.paths.each do |path|
-          flags << " -L#{path}"
+        # First, check pkg-config for the pkg-config module name if provided, then
+        # check pkg-config with the lib name, then fall back to -lname
+        if (pkg_config_name = ann.pkg_config) && (flag = pkg_config(pkg_config_name, static_build))
```

Comment:
This is giving more precedence to the pkg-config settings for the lib, instead of giving priority to the CRYSTAL_LIBRARY_PATH as it should. Otherwise having a local package for bdw-gc will override the patched embedded version we ship.

Reply:
yeah, it will, I don't think `CRYSTAL_LIBRARY_PATH` was a good idea because it binds you to the "path" model of library lookup, which is outdated and replaced with pkgconfig (and `PKG_CONFIG_PATH`)

The better solution is `CRYSTAL_LIBRARY_OVERRIDES="gc=/path/to/whatever otherlib=-lwhatever"` which is far more robust and flexible.